packages/theme/src/cli/services/local-storage.ts

```diff
@@ -144,3 +149,8 @@ function assertThemeStoreExists(storage: LocalStorage<ThemeLocalStorageSchema> =
   }
   return themeStore
 }
+
+/** Provides theme store context to each environment during concurrent multi environment execution */
+export async function useThemeStoreContext<T>(store: string, callback: () => Promise<T>): Promise<T> {
```

Comment:
This pattern is also being used in [ConcurrentOutput](https://github.com/Shopify/cli/blob/7e6a0979b7c702402d6798ed02dfe6d5bf90cb7b/packages/cli-kit/src/private/node/ui/components/ConcurrentOutput.tsx#L41-L50)